app/controllers/tasks_controller.rb

```diff
@@ -10,14 +10,26 @@ def create
     redirect_to list_path(@list)
   end
 
+  def show
+    @task = Task.find(params[:id])
+    @list = @task.list_id
+  end
+
   def new
     @list = List.find(params[:list_id])
     @task = Task.new
   end
 
   def update
     @task = Task.find(params[:id])
-    @task.save
+    @list = List.find(@task.list_id)
```

Comment:
Are we able to ask the Task for the list here, instead of fetching from `List` directly?

It's possible that we might fetch a list which we shouldn't be able to access (because it's not related to the task) like this.

Reply:
Hmm, what would that look like in the database? Every task should belong to a list, so I'm not sure if this would return an invalid/nonexistent list.

Reply:
Hmm, what would that look like in the database? Every task should belong to a list, so I'm not sure if this would return an invalid/nonexistent list. @nickcharlton